tt_eager/tensor/tensor_impl.cpp

```diff
@@ -115,6 +115,7 @@ DeviceBuffer allocate_sharded_buffer_on_device(uint32_t buffer_size_bytes, Devic
                                             const Shape& shape, DataType data_type, Layout layout,
                                             std::optional<ShardSpecBuffer> shard_params,
                                             const MemoryConfig& memory_config) {
+    TT_ASSERT(shard_params.has_value(), "Shard params are required for sharded buffer and they were not initialized");
```

Comment:
this looks like the only line changed that `ttnn` folks are tagged on.. this looks fine to me.

Reply:
Yea - there shouldn't be any meaningful functional changes in ttnn - stuff like asserts. Thanks for taking a look!

Reply:
FYI @arakhmati @eyonland @xanderchin, I think Joseph's approval was what I needed. No input required anymore. S